### PR TITLE
[test/plugin] immigrate h2 & struts scenario

### PR DESCRIPTION
**Pull request by arugal**
Please answer these questions before submitting pull request

- Why submit this pull request?
- [ ] Bug fix
- [x] New feature provided
- [ ] Improve performance

- Related issues
#3583

### CODE REVIEW

File: .github/workflows/plugins-test.yaml [hunk 1]
```diff
@@ -53,7 +53,7 @@ jobs:
       - name: Run okhttp 3.0.x-3.14.x (34)
         run: bash test/plugin/run.sh okhttp-scenario
 
-  Spring31x_ES5_Gateway_Avro:
+  Spring31x_ES5_Gateway_Avro_Struts2.5:
```
Comment (kezhenxu94) [inline .github/workflows/plugins-test.yaml]: `.` is not valid as a yaml section name, unless it’s quoted

File: .github/workflows/plugins-test.yaml [hunk 2]
```diff
@@ -333,7 +335,7 @@ jobs:
       - name: Run zookeeper 3.4.x (14)
         run: bash test/plugin/run.sh zookeeper-scenario
 
-  Spring43:
+  Spring43_Struts2.3:
```
Comment (kezhenxu94) [inline .github/workflows/plugins-test.yaml]: Ditto

File: .github/workflows/plugins-test.yaml (hunk 1 above)
Comment (arugal) [reply .github/workflows/plugins-test.yaml]: done

File: .github/workflows/plugins-test.yaml (hunk 2 above)
Comment (arugal) [reply .github/workflows/plugins-test.yaml]: done

File: test/plugin/scenarios/struts2.5-scenario/config/expectedData.yaml [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+            operationId: 0
+            parentSpanId: -1
+            spanId: 0
+            spanLayer: Http
+            startTime: gt 0
+            endTime: gt 0
+            componentId: 15
+            componentName: ''
+            isError: false
+            spanType: Entry
```
Comment (wu-sheng) [inline test/plugin/scenarios/struts2.5-scenario/config/expectedData.yaml]: I prefer to enhance this case, we should check the ref by adding a http call.

Comment (arugal) [reply test/plugin/scenarios/struts2.5-scenario/config/expectedData.yaml]: Done.

**Comment by wu-sheng** [comment]
Strange. Look like checkout failure?

**Comment by wu-sheng** [comment]
@arugal You could rerun from the `Checks` tab.

**Comment by arugal** [comment]
> @arugal You could rerun from the `Checks` tab.

Rerun still failed.

**Comment by kezhenxu94** [comment]
@arugal could you fix the component id in https://github.com/apache/skywalking/blob/master/test/plugin/scenarios/undertow-scenario/config/expectedData.yaml , the component id of Undertow is changed to 84 in #4398 and it will fail then

**Comment by arugal** [comment]
> @arugal could you fix the component id in https://github.com/apache/skywalking/blob/master/test/plugin/scenarios/undertow-scenario/config/expectedData.yaml , the component id of Undertow is changed to 84 in #4398 and it will fail then

Done.

**Comment by arugal** [comment]
@wu-sheng @dmsolr The test has passed, please review.

**Pull request merged**